server/backup_runner.go

```diff
@@ -50,15 +53,78 @@ func (s *BackupRunner) Run(ctx context.Context, wg *sync.WaitGroup) {
 						log.Error("Auto backup runner PANIC RECOVER", zap.Error(err))
 					}
 				}()
-				s.startAutoBackups(ctx, runningTasks, &mu)
+				s.startAutoBackups(ctx, runningBackupTasks, &muBackup)
+				s.downloadBinlogFiles(ctx)
 			}()
 		case <-ctx.Done(): // if cancel() execute
 			return
 		}
 	}
 }
 
-func (s *BackupRunner) startAutoBackups(ctx context.Context, runningTasks map[int]bool, mu *sync.RWMutex) {
+func (s *BackupRunner) downloadBinlogFiles(ctx context.Context) {
+	normal := api.Normal
+	instanceFind := &api.InstanceFind{RowStatus: &normal}
+	instanceList, err := s.server.store.FindInstance(ctx, instanceFind)
+	if err != nil {
+		log.Error("Failed to retrieve instance list", zap.Error(err))
+		return
+	}
+	downloadTimeout := s.backupRunnerInterval / 2
```

Comment:
Why do we need this?

Reply:
This is to ensure the goroutines finish before the next backup runner cycle.
But I second thought about it and this is a bad design. We should just skip this round of downloading if there're still ongoing downloading processes for the current instance.